compiler/rustc_infer/src/infer/opaque_types/mod.rs

```diff
@@ -377,7 +377,7 @@ impl<'tcx> InferCtxt<'tcx> {
 ///
 /// We ignore any type parameters because impl trait values are assumed to
 /// capture all the in-scope type parameters.
-pub struct ConstrainOpaqueTypeRegionVisitor<'tcx, OP: FnMut(ty::Region<'tcx>)> {
+struct ConstrainOpaqueTypeRegionVisitor<'tcx, OP: FnMut(ty::Region<'tcx>)> {
     pub tcx: TyCtxt<'tcx>,
```

Comment:
now that the type is private, it's fields can be as well. Also... interesting that we don't have a lint for this case

Reply:
The `unreachable_pub` lint was deliberately changed to *not* warn on this case in #126040, because it was deemed annoying. Specifically:

> This PR restrict the `unreachable_pub` lint by not linting on `pub` fields of `pub(restricted)` structs and unions. This is done because that can quickly clutter the code for an uncertain value, in particular since the "real" visibility is defined by the parent (the struct it-self).

The implementation does more than that description indicates -- it also doesn't lint on `pub` fields of non-`pub` structs and unions. Perhaps it should?